src/main/scala/mesosphere/marathon/core/leadership/LeaderShipModule.scala

```diff
@@ -0,0 +1,33 @@
+package mesosphere.marathon.core.leadership
+
+import akka.actor.{ ActorRef, ActorRefFactory, Props }
+import mesosphere.marathon.core.leadership.impl.{
+  ActorLeadershipCoordinator,
+  LeadershipCoordinationActor,
+  WhenLeaderActor
+}
+
+class LeadershipModule(actorRefFactory: ActorRefFactory) {
+
+  private[this] var whenLeaderRefs = Set.empty[ActorRef]
+  private[this] var started: Boolean = false
+
+  def startWhenLeader(props: => Props, name: String, considerPreparedOnStart: Boolean = true): ActorRef = {
```

Comment:
The effect of considerPreparedOnStart is not clear by only reading the signature.
please add some documentation.
BTW: Would it be too much of an overhead to always send a reply to confirm the start?